### PR TITLE
Issue-654, Prohibiting cluster spec updating until they are running

### LINKED ISSUE #654: Prohibit update operation until a cluster has `NO_OPERATION` state
We need to prevent all updating operations until a cluster is in the "RUNNING" state and/or has "NO_OPERATION". 
For example, there can be an error when you try to resize a cluster and change some of the config fields at the same time

### PR DESCRIPTION
**Pull request by worryg0d**
The PR adds validation to the update webhooks to prevent cluster spec updating until their state is `running` or there is no cluster operation.

Also, it cleans up the codebase and adds some minor fixes.

closes #654

### CODE REVIEW

File: apis/clusters/v1beta1/cadence_webhook.go [hunk 1]
```diff
@@ -255,6 +259,11 @@ func (cv *cadenceValidator) ValidateUpdate(ctx context.Context, old runtime.Obje
 		}
 	}
 
+	// ensuring if the cluster is ready for the spec updating
+	if (c.Status.CurrentClusterOperationStatus != models.NoOperation || c.Status.State != models.RunningStatus) && c.Generation != oldCluster.Generation {
```
Comment (DoodgeMatvey) [inline apis/clusters/v1beta1/cadence_webhook.go:263]: Code should be self documented, I don't think that we need such comments

File: apis/clusters/v1beta1/cassandra_webhook.go [hunk 2]
```diff
@@ -195,6 +195,11 @@ func (cv *cassandraValidator) ValidateUpdate(ctx context.Context, old runtime.Ob
 		return fmt.Errorf("cannot update immutable fields: %v", err)
 	}
 
+	// ensuring if the cluster is ready for the spec updating
+	if (c.Status.CurrentClusterOperationStatus != models.NoOperation || c.Status.State != models.RunningStatus) && c.Generation != oldCluster.Generation {
+		return models.ErrClusterIsNotReadyToUpdate
+	}
+
```
Comment (DoodgeMatvey) [inline apis/clusters/v1beta1/cassandra_webhook.go:202]: Do we really need this check? `c.Generation != oldCluster.Generation`
We already have it in `SetupWithManager` and there will be no update in false case. 
And check all such cases
`	if event.ObjectNew.GetGeneration() == event.ObjectOld.GetGeneration() {
					return false
				}`

**Review by ribaraka — APPROVED** [review]

**Review by DoodgeMatvey — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/cassandra_webhook.go (hunk 2 above)
Comment (worryg0d) [reply apis/clusters/v1beta1/cassandra_webhook.go:202]: The following logic prohibits updating the resource spec field. Validating webhooks trigger to update requests only on `Patch` and `Update` operation. However, we need to prevent any updations of the `spec`, so we have to understand what triggered an update. If it changes the `spec` it also changes the `metadata.generation` field. If we don't ensure that this is a `spec` update then it will denies any `Update` requests, but controllers of the following resources also update resources by adding/deleting annotations/finalizers by using `client.Update` and `client.Patch`.

In summary, It helps to understand if it was an update of the resource `spec` field or any other update, but we have to prohibit only updates of the `spec`.

File: apis/clusters/v1beta1/cadence_webhook.go (hunk 1 above)
Comment (worryg0d) [reply apis/clusters/v1beta1/cadence_webhook.go:263]: I discussed it with teammates and we decided to keep it as it is because it enhances the readability of the following check

Comment (DoodgeMatvey) [reply apis/clusters/v1beta1/cadence_webhook.go:263]: Strange to write comment for check if no operation or not running status, but ok

Comment (worryg0d) [reply apis/clusters/v1beta1/cadence_webhook.go:263]: This comment refers to spec update only, no the general k8s resource update

File: apis/clusters/v1beta1/cassandra_webhook.go (hunk 2 above)
Comment (DoodgeMatvey) [reply apis/clusters/v1beta1/cassandra_webhook.go:202]: Validating webhook can validate the request and potentially reject it based on our business logic.
During this phase, the Generation field of the resource has not yet been incremented because the update has not been persisted. So, why should we check this?

**Review by DoodgeMatvey — CHANGES_REQUESTED** [review]

File: apis/clusters/v1beta1/cassandra_webhook.go (hunk 2 above)
Comment (worryg0d) [reply apis/clusters/v1beta1/cassandra_webhook.go:202]: For the incoming new object it is already changed, because this is a part of it and it is changes only by k8s

Comment (DoodgeMatvey) [reply apis/clusters/v1beta1/cassandra_webhook.go:202]: Ok

**Review by DoodgeMatvey — APPROVED** [review]

**Review by tengu-alt — APPROVED** [review]
